src/main/java/com/fundation/logic/model/SearchVideo.java

```diff
@@ -17,49 +18,75 @@
+            if (fileExtractor.isHidden() == criteria.getFileHidden())
+            {
+                if ((fileExtractor.canRead() && !fileExtractor.canWrite()) == criteria.getFileReadOnly())
+                {
+                    if (compareFile(fileExtractor)!=null){
+                    listFileAndDirectory.add(compareFile(fileExtractor));
+                    }
                 }
-            }
-            if (!(criteria.getPath().equals("")) && (!criteria.getFileName().equals("")) && (!criteria.getExtension().equals(""))) {
-                if (fileExtractor.getName().contains(criteria.getFileName()) && fileExtractor.getName().contains(criteria.getExtension())) {
-                    listFileAndDirectory.add(new File(fileExtractor.getAbsolutePath()));
+                else
+                {
+                    if (compareFile(fileExtractor)!=null){
+                    listFileAndDirectory.add(compareFile(fileExtractor));
+                    }
                 }
             }
-            if (!(criteria.getPath().equals("")) && (criteria.getFileName().equals("")) && (!criteria.getExtension().equals(""))) {
-                if (fileExtractor.getName().contains(criteria.getExtension())) {
-                    listFileAndDirectory.add(new File(fileExtractor.getAbsolutePath()));
+            else
+            {
+                if ((fileExtractor.canRead() && !fileExtractor.canWrite()) == criteria.getFileReadOnly())
+                {
+                    if (compareFile(fileExtractor)!=null){
+                    listFileAndDirectory.add(compareFile(fileExtractor));
+                    }
+                }
+                else
+                {   if (compareFile(fileExtractor)!=null){
+                    listFileAndDirectory.add(compareFile(fileExtractor));
+                    }
                 }
             }
         }
         return listFileAndDirectory;
     }
-}
-
+    private File compareFile(File fileExtractor){
+        if (!(criteria.getPath()==null) && (criteria.getFileName() == null) && (criteria.getExtension() == null)) {
+            return (new File(fileExtractor.getAbsolutePath()));
+        }
+        if (!(criteria.getPath() == null) && (!(criteria.getFileName() == null)) && (criteria.getExtension() == null)) {
+            if (fileExtractor.getName().contains(criteria.getFileName()) && fileExtractor.getPath().contains(criteria.getPath())) {
+                return (new File(fileExtractor.getAbsolutePath()));
+            }
+        }
+        if (!(criteria.getPath() == null) && (!(criteria.getFileName() == null)) && (!(criteria.getExtension() == null))) {
+            if (fileExtractor.getName().contains(criteria.getFileName()) && fileExtractor.getName().contains(criteria.getExtension())) {
+                return (new File(fileExtractor.getAbsolutePath()));
+            }
+        }
+        if (!(criteria.getPath() == (null)) && (criteria.getFileName()==null) && (!(criteria.getExtension() == null))) {
```

Comment:
space before and after the logic operator "=="

Reply:
that was fixed.